### DISCUSSION TITLE
Dishwasher not supported?

### DISCUSSION THREAD

**Discussion by Zristi** [thoukydides/homebridge-homeconnect#131] [Q&A]
Hi, I'm getting the following message:
Bosch Dishwasher (E-Nr: SPV6YMX11E/08)
This appliance does not support any programs.

Does that mean my dishwasher is not supported?

Thanks!

**Comment by thoukydides** [comment]
This plugin does not special-case individual appliances - all dishwashers are treated identically, with the individual capabilities determined via the Home Connect API.

The ```Does not support any programs``` message is logged if the Home Connect API does not list any supported programs when queried - either returning an empty list or an error. There are several possible reasons for this, e.g.:
* The appliance does not support selecting/starting any programs via the Home Connect API. This is unlikely to be the case for a dishwasher.
* The appliance was not in a suitable state to return a list of its supported programs. Some appliances restrict the functionality supported via the Home Connect API when they require maintenance, such as adding consumables or emptying waste containers. Again, this is unlikely to be the case for a dishwasher.
* The Home Connect API encountered an error - either a problem within the cloud services, or connecting to the appliance. This is the most likely case.

Details of the issue should have been recorded in the Homebridge log file, a few lines above the message you reported.

More details of API requests and responses will be logged if you run Homebridge in debug mode (using the `-D` command line switch).

If there was an error then restarting Homebridge, or performing a HomeKit `Identify` operation on the accessory, will try to read the supported programs again.
